File: ruby/collatz-conjecture/collatz_conjecture.rb
module CollatzConjecture
  def self.steps(number)
    raise ArgumentError unless number > 0
    steps = 0

    until number == 1
      if number.odd?
        number = number * 3 + 1
      else
        number /= 2
      end
      steps += 1
    end

    steps
  end
end

File: python/collatz-conjecture/collatz_conjecture.py
def collatz_steps(number):
    if number < 1:
        raise ValueError('Raise error')
    steps = 0

    while number != 1:
        if number % 2 == 1:
            number = number * 3 + 1
        else:
            number /= 2
        steps += 1

    return steps


